components/style/values/generics/svg.rs

```diff
@@ -101,14 +102,44 @@ impl<ColorType: Parse, UrlPaintServer: Parse> Parse for SVGPaint<ColorType, UrlP
 /// https://www.w3.org/TR/SVG11/painting.html#StrokeProperties
 #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
 #[derive(Clone, Copy, Debug, PartialEq, ToCss, HasViewportPercentage)]
-#[derive(ToComputedValue, ToAnimatedValue, ComputeSquaredDistance)]
+#[derive(ToComputedValue, ToAnimatedValue)]
 pub enum SvgLengthOrPercentageOrNumber<LengthOrPercentageType, NumberType> {
     /// <length> | <percentage>
     LengthOrPercentage(LengthOrPercentageType),
     /// <number>
     Number(NumberType),
 }
 
+impl<L, N> ComputeSquaredDistance for SvgLengthOrPercentageOrNumber<L, N>
+    where NumberOrPercentage: From<L> + From<N> + ComputeSquaredDistance,
+          L: ComputeSquaredDistance + Copy,
+          N: ComputeSquaredDistance + Copy
```

Comment:
The indentation is all wrong here.

```rust
where
    NumberOrPercentage: From<L> + From<N> + ComputeSquaredDistance,
    L: ComputeSquaredDistance + Copy,
    N: ComputeSquaredDistance + Copy,
```

It would also be better to use `Into<T>`:

```rust
where
    NumberOrPercentage: ComputeSquaredDistance,
    L: ComputeSquaredDistance + Copy + Into<NumberOrPercentage>,
    N: ComputeSquaredDistance + Copy + Into<NumberOrPercentage>,
```

Finally, given `NumberOrPercentage` is just a concrete type, its trait bounds are useless:

```rust
where
    L: ComputeSquaredDistance + Copy + Into<NumberOrPercentage>,
    N: ComputeSquaredDistance + Copy + Into<NumberOrPercentage>,
```